api/v1beta1/heat_webhook.go

```diff
@@ -94,8 +99,30 @@ func (r *Heat) ValidateCreate() error {
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Heat) ValidateUpdate(old runtime.Object) error {
 	heatlog.Info("validate update", "name", r.Name)
+	oldHeat, ok := old.(*Heat)
+	if !ok {
+		return fmt.Errorf("Expected a Heatv1 object, but got %T", oldHeat)
```

Comment:
maybe should return a `apierrors.NewInternalError( fmt.Errorf())` 
```
NewInternalError returns an error indicating the item is invalid and cannot be processed.
```

Reply:
Yeah, sounds good. Changed it.